pkg/util/api.go

```diff
@@ -91,6 +96,15 @@ func BytesToObject(bytes []byte, object interface{}) error {
 	if err := unmarshal(bytes, object); err != nil {
 		return err
 	}
+
+	if err := interpret(object); err != nil {
+		return &HTTPError{
+			ErrorType:   "BadRequest",
+			Description: "Unable to decode TLS credentials: " + err.Error(),
```

Comment:
Should be "Could not interpret", not "unable to decode TLS credentials".

Also: do you really need the additional `HTTPError`  here? Why not return the error from `interpret()`?